File: Python/Sorteador/Sorteador.py
import time
import sys
from random import randint

listaDeCoisas = []
item = ''

while item != "0":   
    item = input("Digite o item para adicionar a lista ou ZERO para sair: ")

    if item == "0":
        break
    else :
        
        if item in listaDeCoisas:
            print(f"Item {item} encontrado na lista.............")
            listaDeCoisas.remove(item)

        else:
            listaDeCoisas.append(item)
            print(f"Item {item} adicionado a lista")


        print("Lista ate agora:")
        if len(listaDeCoisas) > 0:
            for item in listaDeCoisas:
                print(item)
            # print(f'Lista ate agora: {listaDeCoisas}\n')
        else:
            print("Lista vazia")



texto = "Sorteando item ....."
for letra in texto:
    sys.stdout.write(letra)
    sys.stdout.flush()
    time.sleep(0.1)
print()

if randint(0,1) == 1:
    print("Fazer amorzinho gostoso hmmmm")
else:
    i = randint(0, len(listaDeCoisas) - 1)
    print(f'Item sorteado: {listaDeCoisas[i]}')

File: Z - Pojects e Temp/Projeitos/Sorteador/sorteador.js
let listaDeCoisas = [];

function adicionarItem() {
    const itemInput = document.getElementById('item');
    const item = itemInput.value.trim();
    const resultado = document.getElementById('resultado');
    
    if (item === '') {
        //alert('Por favor, digite um item válido!');
        return;
    }

    if (listaDeCoisas.includes(item)) {
        listaDeCoisas = listaDeCoisas.filter(i => i !== item);
        resultado.textContent = `Item "${item}" removido da lista!`;
    } else {
        listaDeCoisas.push(item);
        resultado.textContent = `Item "${item}" adicionado à lista!`;
    }

    // Remove itens vazios da lista
    listaDeCoisas = listaDeCoisas.filter(item => item !== '');

    atualizarLista();
    itemInput.value = "";
}

function atualizarLista() {
    const lista = document.getElementById('lista');
    lista.innerHTML = '';
    
    listaDeCoisas.forEach(item => {
        const li = document.createElement('li');
        li.textContent = item;
        lista.appendChild(li);
    });
}

function sortear() {
    if (listaDeCoisas.length === 0) {
        alert('Adicione itens antes de sortear!');
        return;
    }

    const resultado = document.getElementById('resultado');
    resultado.textContent = 'Sorteando...';

    // Efeito de "sorteando"
    setTimeout(() => {
        // if (Math.random() < 0.5) {
        //     resultado.textContent = 'Vamos fazer amorzinho gostoso!!';
        //     resultado.innerHTML += '<br>HMMMMMM!!!';
        // } else {
            const indice = Math.floor(Math.random() * listaDeCoisas.length);
            resultado.textContent = `Item sorteado: ${listaDeCoisas[indice]}`;
        // }
    }, 1000);
}

// Adiciona event listener para a tecla Enter
document.getElementById('item').addEventListener('keypress', function(event) {
    if (event.key === 'Enter') {
        adicionarItem();
    }
});